scripts/pinned_build.sh

```diff
@@ -72,31 +72,19 @@ install_dependencies() {
         else
             unset SUDO_CMD
         fi
-        try $SUDO_CMD apt-get update
-        try $SUDO_CMD apt-get update --fix-missing
-        export DEBIAN_FRONTEND='noninteractive'
-        TZ='Etc/UTC'
-        try $SUDO_CMD apt-get install -y tzdata
-        try $SUDO_CMD apt-get install -y \
-            build-essential \
-            bzip2 \
-            cmake \
-            curl \
-            file \
-            git \
-            libbz2-dev \
-            libgmp-dev \
-            libncurses5 \
-            libssl-dev \
-            libtinfo-dev \
-            libzstd-dev \
-            python3 \
-            python3-numpy \
-            time \
-            unzip \
-            wget \
-            zip \
-            zlib1g-dev
+        DEPENDENCIES="$(cat "$SCRIPT_DIR/pinned_deps.txt")"
+        echo 'Checking for missing package dependencies.'
+        dpkg -l $DEPENDENCIES &> /dev/null && MISSING_DEPS='false' || MISSING_DEPS='true'
```

Comment:
you can put the command directly inside the `if`, so you don't need a `MISSING_DEPS` variable.

This doesn't look sufficient, as just installing and uninstalling the package still returns 0 exit code, for example
```
# dpkg -l cmake ; echo $?
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
un  cmake          <none>       <none>       (no description available)
0

```

Reply:
This is _really_ good feedback! I switched from `dpkg -l` to `dpkg -s`, which I verified does not have the problematic behavior on Ubuntu 22.04:

![2023-03-20 15-55-23 - dpkg -s package](https://user-images.githubusercontent.com/34947245/226454689-9359dad0-b5d0-4a9a-bf58-5e0906beef9e.png)

Thank you.